bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/Utils.java

```diff
@@ -256,8 +284,102 @@ public static Field createAdditionalField(RecordType recordType, String fieldNam
         }
     }
 
+    private static Object validateAndGetTableArrayValue(TomlTableArrayNode tomlNode, Set<TomlNode> visitedNodes,
```

Comment:
Lets move this to compile time. May be we can do with a seperate PR.

Reply:
I'll address this in issue https://github.com/ballerina-platform/ballerina-lang/issues/31889

Reply:
I'll address this in issue https://github.com/ballerina-platform/ballerina-lang/issues/31889

Reply:
I'll address this in issue https://github.com/ballerina-platform/ballerina-lang/issues/31889

Reply:
I'll address this in issue https://github.com/ballerina-platform/ballerina-lang/issues/31889

Reply:
I'll address this in issue https://github.com/ballerina-platform/ballerina-lang/issues/31889

Reply:
I'll address this in issue https://github.com/ballerina-platform/ballerina-lang/issues/31889